crawl.sh

```
@@ -0,0 +1,324 @@
+OUTPUT_DIR="$(pwd)/output"
+CACHE_DIR="$(pwd)/cache"
+LOCK_FILE="/tmp/crawler_to_md.lock"
+PID_FILE="/tmp/crawler_to_md.pid" # Process ID file for additional lock mechanism
+LOCK_TIMEOUT=$((30 * 60)) # 30 minutes in seconds
+
+# Function to check if a process with given PID exists and is our script
+is_process_running() {
+  local pid=$1
+  if ps -p "$pid" > /dev/null; then
+    # Get the command line of the process to verify it's our crawler script
+    local cmd_line=$(ps -p "$pid" -o args= 2>/dev/null || ps -p "$pid" -o command= 2>/dev/null)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)